src/lib/operators/maintenance/drop_index.cpp

```diff
@@ -0,0 +1,82 @@
+#include "drop_index.hpp"
+
+#include <sstream>
+
+#include "constant_mappings.hpp"
+#include "hyrise.hpp"
+#include "operators/insert.hpp"
+#include "storage/table.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+#include "utils/assert.hpp"
+
+
+namespace opossum {
+
+DropIndex::DropIndex(const std::string& init_index_name,
+                         const std::string& init_target_table_name,
+                         const std::shared_ptr<const AbstractOperator>& input_operator)
+    : AbstractReadWriteOperator(OperatorType::DropIndex, input_operator),
+      index_name(init_index_name),
+      target_table_name(init_target_table_name)
+    {}
+
+const std::string& DropIndex::name() const {
+  static const auto name = std::string{"DropIndex"};
+  return name;
+}
+
+std::string DropIndex::description(DescriptionMode description_mode) const {
+  // TODO: craft usefull description print output
+  /*
+  std::ostringstream stream;
+
+  const auto* const separator = description_mode == DescriptionMode::SingleLine ? ", " : "\n";
+
+  // If the input operator has already been cleared, we cannot retrieve its columns anymore. However, since the table
+  // has been created, we can simply pull the definitions from the new table.
+  const auto column_definitions = left_input_table()
+                                  ? left_input_table()->column_definitions()
+                                  : Hyrise::get().storage_manager.get_table(table_name)->column_definitions();
+
+  stream << AbstractOperator::description(description_mode) << " '" << table_name << "' (";
+  for (auto column_id = ColumnID{0}; column_id < column_definitions.size(); ++column_id) {
+    const auto& column_definition = column_definitions[column_id];
+
+    stream << "'" << column_definition.name << "' " << column_definition.data_type << " ";
+    if (column_definition.nullable) {
+      stream << "NULL";
+    } else {
+      stream << "NOT NULL";
+    }
+
+    if (column_id + 1u < column_definitions.size()) {
+      stream << separator;
+    }
+  }
+  stream << ")";
+
+  return stream.str();
+   */
+  return std::string("DROP INDEX description");
+}
+
+std::shared_ptr<const Table> DropIndex::_on_execute(std::shared_ptr<TransactionContext> context) {
+  auto table_to_be_indexed = Hyrise::get().storage_manager.get_table(target_table_name);
```

Comment:
You introduce a table wrapper operator in field "input_operator", which also holds the table after beeing executed.. why not use it here?